frontend/connection/src/types.ts

```diff
@@ -62,6 +62,30 @@ export interface StreamlitEndpoints {
    */
   setStaticConfigUrl(url: string): void
 
+  /**
+   * Send postMessage to host with client errors
+   * @param component component causing the error
+   * @param customComponentName if custom component, component's name
+   * @param error error status code or message
+   * @param message additional error info
+   * @param source component src (url)
+   */
+  sendClientErrorToHost(
+    component: string,
+    customComponentName: string,
+    error: string | number,
+    message: string,
+    source: string
+  ): void
+
+  /**
+   * Checks if the custom component src has successful response.
+   * If not, sends CLIENT_ERROR message with error info.
+   * @param source The source to check.
+   * @param componentName The component for which the source is being checked.
+   */
+  checkSourceResponse(source: string, componentName?: string): Promise<void>
```

Comment:
Ah, I see the definition here. Since this is specific to custom component, `checkComponentSourceResponse`?

Reply:
May need to leverage it for other media (ex: video subtitles `<track>` has no onerror event) so kept function name more general for now

Reply:
Awesome. Source is supposed to be a url right, might just make sense to make that more explicit in the name then. `checkSourceUrlResponse(sourceUrl: string...`